nullaway/src/main/java/com/uber/nullaway/fixserialization/out/ClassAndMemberInfo.java

```diff
@@ -33,20 +35,30 @@
 /** Class and member corresponding to a program point at which an error / fix was reported. */
 public class ClassAndMemberInfo {
   /** Path to the program point of the reported error / fix */
-  public final TreePath path;
+  @Nullable public TreePath path;
 
   // Finding values for these properties is costly and are not needed by default, hence, they are
   // not final and are only initialized at request.
   @Nullable private Symbol member;
 
-  @Nullable private ClassTree clazz;
+  @Nullable private Symbol.ClassSymbol clazz;
 
   public ClassAndMemberInfo(TreePath path) {
+    Preconditions.checkNotNull(path);
     this.path = path;
   }
 
+  public ClassAndMemberInfo(Tree regionTree) {
+    this.member = ASTHelpers.getSymbol(regionTree);
```

Comment:
Do we want to have some form of sanity check here that this is a member of a class (method or field) vs any other possible symbol? Maybe even check that, once we do get the class symbol, `this.member` is a member of the class, rather than something more deeply nested.

Reply:
@lazaroclapp not exactly sure what you are asking for here.  I would think a good assertion would be that `regionTree` is in fact a `VariableTree` for a field or a `MethodTree`.  But that doesn't sound like what you are thinking about.  Based on the code it seems that `this.member` must be a member of `this.clazz`?

Reply:
I added a check in ee15adca2f44aa33b557b0e733b01352840486d5 that I think is what we are looking for.  But @lazaroclapp you may want to do another test run once this lands before cutting a release, in case the check unexpectedly fails

Reply:
I think this should be enough for now. Though I am not sure how this stuff interacts with records/annotations/inferfaces/enums/etc and whether or not there exists a case such that:

1. `regionTree` and thus `this.member` represent a method or field.
2. `clazz = ASTHelpers.enclosingClass(this.member)` is such that `this.member` isn't an element of  `clazz.members()` (because we skipped some construct in between with a scope that it _is_ a member of)

Reply:
Yeah I'm not 100% sure either.  We can try to write some more tests in a follow up